homeassistant/components/kodi/media_player.py

```diff
@@ -370,40 +368,20 @@ def state(self) -> MediaPlayerState:
         if self._no_active_players:
             return MediaPlayerState.IDLE
 
-        if self._properties["speed"] == 0:
+        # On first connect, if Kodi already plays any media,
+        # self_properties might be empty
+        if self._properties and self._properties["speed"] == 0:
             return MediaPlayerState.PAUSED
 
         return MediaPlayerState.PLAYING
 
     async def async_added_to_hass(self) -> None:
-        """Connect the websocket if needed."""
-        if not self._connection.can_subscribe:
-            return
-
-        if self._connection.connected:
-            await self._on_ws_connected()
-
-        async def start_watchdog(event=None):
-            """Start websocket watchdog."""
-            await self._async_connect_websocket_if_disconnected()
-            self.async_on_remove(
-                async_track_time_interval(
-                    self.hass,
-                    self._async_connect_websocket_if_disconnected,
-                    WEBSOCKET_WATCHDOG_INTERVAL,
-                )
-            )
-
-        # If Home Assistant is already in a running state, start the watchdog
-        # immediately, else trigger it after Home Assistant has finished starting.
-        if self.hass.state == CoreState.running:
-            await start_watchdog()
-        else:
-            self.hass.bus.async_listen_once(EVENT_HOMEASSISTANT_STARTED, start_watchdog)
+        """Register callbacks for needed api methods."""
+        await self._connman.add_callback_on_connect(self._on_ws_connected)
+        await self._connman.add_callback_on_disconnect(self._on_ws_disconnected)
 
     async def _on_ws_connected(self):
-        """Call after ws is connected."""
-        self._connect_error = False
+        """Call after websocket is connected."""
         self._register_ws_callbacks()
 
         version = (await self._kodi.get_application_properties(["version"]))["version"]
```

Comment:
I'm not sure this belongs in the media player after the refactoring

Reply:
I suppose you mean updating the device with the Kodi version?!
At the moment the Kodi device is automatically registered through the media_player entity. Since the KodiConnectionManager is initialized before the media_player entity is add to HA, I cannot update the device registry when the Kodi component is configured in HA for the first time.
I would need to manually register the device in the KodiConnectionManager. Would you go for that?

Reply:
we can set the device info from the base class

Reply:
I did both now. And tested adding and removing from HA.